Rewrite of the CSI driver docs. (#2898)
MicrosoftDocs/azure-stack-docs: azure-managed-lustre/use-csi-driver-kubernetes.md

1. [Create virtual network peering](#create-virtual-network-peering)

1. [Install the Azure Managed Lustre CSI Driver for Kubernetes](#install-the-csi-driver).

1. [Create and configure a persistent volume](#create-and-configure-a-persistent-volume).

1. [Check the installation](#check-the-installation) by optionally using an echo pod to confirm the driver is working.

The following sections describe each task in greater detail.

## Create an Azure Managed Lustre file system

If you haven't already created your Azure Managed Lustre file system cluster, create the cluster now. For instructions, see [Create an Azure Managed Lustre file system in the Azure portal](create-file-system-portal.md). Currently, the driver can only be used with a existing Azure Managed Lustre file system.

## Create an AKS Cluster

If you haven't already created your AKS cluster, create a cluster deployment. See [Deploy an Azure Kubernetes Service (AKS) cluster](/azure/aks/learn/quick-kubernetes-deploy-portal).

## Create virtual network peering

> [!NOTE]
> Skip this network peering step if you installed AKS into a subnet on the Azure Managed Lustre virtual network.

> The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the **Properties** blade and finding the **Infrastructure** resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_&lt;aks-rg-name&rt;\_&lt;aks-cluster-name&rt;\_&lt;region&rt;**.

Reviewer: ```suggestion
The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the **Properties** blade and finding the **Infrastructure** resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_\<aks-rg-name\>\_\<aks-cluster-name\>\_\<region\>**.
```
The right side angle brackets weren't displaying correctly so I've made changes to fix them... but @jeffbearer - please check that this is the correct pattern: 

MC\_\<aks-rg-name\>\_\<aks-cluster-name\>\_\<region\>

Consult [Virtual Network Peering](/azure/virtual-network/virtual-network-peering-overview) to peer the AKS virtual network with your Azure Manages Lustre virtual network.

> > [TIP]

Reviewer: ```suggestion
> [!TIP]
```

> Due to the naming of the MC_ resource groups and virtual networks, names of networks can be similar or the same across multiple AKS deployments. When setting up peering pay close attention that you are choosing the AKS networks that you intend to choose.

## Connect to the AKS cluster

Connect to the Azure Kubernetes Service cluster by doing these steps:

> 1. Open a terminal session with access to the Azure CLI tools, and login to your Azure account.

Reviewer: ```suggestion
1. Open a terminal session with access to the Azure CLI tools and log in to your Azure account.
```

---

# Use the Azure Managed Lustre CSI Driver with Azure Kubernetes Service

> This article describes how to plan, install, and use [Azure Managed Lustre](/azure/azure-managed-lustre) in [Azure Kubernetes Service (AKS)](/azure/aks/) with the [Azure Managed Lustre Kubernetes container support interface driver (Azure Managed Lustre CSI Driver)](https://github.com/kubernetes-sigs/azurelustre-csi-driver).

Author: incorporated

## About the Azure Managed Lustre CSI driver for AKS

> The Azure Managed Lustre Continer Support Interface (CSI) Driver for AKS enables you to access Azure Managed Lustre storage as persistant storage volumes from Kubernetes containers deployed in [Azure Kubernetes Service (AKS)](/azure/aks/).

Reviewer 2: ```suggestion
The Azure Managed Lustre Continer Support Interface (CSI) driver for AKS enables you to access Azure Managed Lustre storage as persistant storage volumes from Kubernetes containers deployed in [Azure Kubernetes Service (AKS)](/azure/aks/).
```

## Compatible Kubernetes versions

> The Azure Managed Lustre CSI Driver for AKS is compatible with [Azure Kubernetes Service (AKS)](/azure/aks/). Other Kubernetes installations are not currently supported.

Reviewer 2: ```suggestion
The Azure Managed Lustre CSI driver for AKS is compatible with [Azure Kubernetes Service (AKS)](/azure/aks/). Other Kubernetes installations are not currently supported.
```

AKS Kubernetes versions 1.21 and later are supported. This includes all versions currently available when creating a new AKS cluster.

> [!IMPORTANT]
> The Azure Managed Lustre CSI driver only works with the Ubuntu Linux OS SKU for Node Pools of AKS currently.

## Compatible Lustre versions

> The Azure Managed Lustre CSI Driver for AKS is compatible with [Azure Managed Lustre](/azure/azure-managed-lustre). Other Lustre installations are not currently supported.  

Author: incorporated

> The Azure Managed Lustre CSI Driver versions 0.1.10 and later are supported with the current version of the Azure Managed Lustre Service.

Author: incorporated

<!---
See [Upgrading Azure Managed Lustre CSI driver](/azure/azure-managed-lustre/csi-driver-upgrading) for more info on upgrading the CSI driver.
--->

## Prerequisites

> Before you deploy the Azure Managed Lustre CSI Driver for AKS, complete the following prerequisites:

Author: incorporated

- An Azure account with an active subscription. [Create an account for free](https://azure.microsoft.com/free/).
- A terminal environment with the Azure CLI tools installed. See [Get started with Azure CLI](cli/azure/get-started-with-azure-cli)
- kubectl the kubernetes management tool is installed in your terminal environment. See [Quickstart: Deploy an Azure Kubernetes Service (AKS) cluster using Azure CLI](azure/aks/learn/quick-kubernetes-deploy-cli#connect-to-the-cluster)
- Create an Azure Managed Lustre deployment. See [Azure Managed Lustre File System documentation](/azure/azure-managed-lustre)

## Plan your AKS Deployment

There are several different options when deploying Azure Kubernetes Service that affect the operation between AKS and Azure Managed Lustre.

### Determine the network type to use with AKS

> There are two network types that are compatible with the Ubuntu Linux OS SKU, kubenet and the Azure Container Network Interface (CNI) Driver. Both options work with the Azure Managed Lustre CSI Driver for AKS but they have different requirements that need to be understood when setting up virtual networking and AKS. See [Networking concepts for applications in Azure Kubernetes Service (AKS)](/azure/aks/concepts-network) for more information on determining the proper selection.

Author: incorporated

- Install AKS into it's own Virtual Network and create a virtual network peering with the Azure Managed Lustre Virtual Network.
- Use *Bring your Own Networking* option in AKS to install AKS on a new subnet on the Azure Managed Lustre Virtual Network.

> [!NOTE]
> Installing AKS onto the same subnet as Azure Managed Lustre is not recommended.

### Peering AKS and Azure Managed Lustre virtual networks

The option to peer two different virtual networks has the advantage of separating the management of the various networks to different privileged roles. Peering can also provide additional flexibility as it can be made across Azure subscriptions or regions. Virtual Network Peering will require coordination between the two networks to avoid choosing conflicting IP network spaces.

![Diagram showing two VNets, one for Azure Managed Lustre and one for AKS, with a VNet peering arrow connecting them.](media/use-csi-driver-kubernetes/subnet-access-option-2.png)

> ### Installing AKS into a Subnet on the Azure Managed Lustre Virtual Network

Reviewer 2: ```suggestion
### Installing AKS into a subnet on the Azure Managed Lustre virtual network
```

Author: incorporated

> The option to install the AKS cluster into the Azure Managed Lustre Virtual Network with the *Bring Your Own Network* feature in AKS can be advantageous where you want scenarios where the network is managed singularly. An additional subnet sized to meet your AKS networking requirements will need to be created in the Azure Managed Lustre virtual network.

Reviewer 2: ```suggestion
The option to install the AKS cluster into the Azure Managed Lustre virtual network with the *Bring Your Own Network* feature in AKS can be advantageous where you want scenarios where the network is managed singularly. An additional subnet sized to meet your AKS networking requirements will need to be created in the Azure Managed Lustre virtual network.
```

Author: incorporated

There is no privilege separation for network management when provisioning AKS onto the Azure Managed Lustre Network and the AKS service principal will need privileges on the Azure Managed Lustre virtual network.

![Diagram showing Azure Managed Lustre VNet with two subnets, one for the Lustre file system and one for AKS.](media/use-csi-driver-kubernetes/subnet-access-option-1.png)

## Setup overview

To enable the Azure Managed Lustre CSI Driver for Kubernetes, perform these steps:

1. [Create an Azure Managed Lustre file system](#create-an-azure-managed-lustre-file-system)

1. [Create an AKS Kubernetes Cluster](#create-an-aks-kubernetes-cluster)

1. [Create virtual network peering](#create-virtual-network-peering)

1. [Install the Azure Managed Lustre CSI Driver for Kubernetes](#install-the-csi-driver).

1. [Create and configure a persistent volume](#create-and-configure-a-persistent-volume).

1. [Check the installation](#check-the-installation) by optionally using an echo pod to confirm the driver is working.

The following sections describe each task in greater detail.

## Create an Azure Managed Lustre file system

> If you haven't already created your Azure Managed Lustre File System cluster, create the cluster now. For instructions, see [Create an Azure Managed Lustre file system in the Azure portal](create-file-system-portal.md). Currently, the driver can only be used with a pre-existing Azure Managed Lustre system.

Reviewer 2: ```suggestion
If you haven't already created your Azure Managed Lustre file system cluster, create the cluster now. For instructions, see [Create an Azure Managed Lustre file system in the Azure portal](create-file-system-portal.md). Currently, the driver can only be used with an existing Azure Managed Lustre file system.
```

Author: incorporated

> ## Create an AKS Kubernetes Cluster

Reviewer 2: ```suggestion
## Create an AKS cluster
```

Author: incorporated

> If you haven't already created your AKS cluster, create a cluster deployment, see [Deploy an Azure Kubernetes Service (AKS) cluster](/azure/aks/learn/quick-kubernetes-deploy-portal).

Reviewer 2: ```suggestion
If you haven't already created your AKS cluster, create a cluster deployment. See [Deploy an Azure Kubernetes Service (AKS) cluster](/azure/aks/learn/quick-kubernetes-deploy-portal).
```

## Create virtual network peering

> ^[NOTE]

Reviewer 2: ```suggestion
> [!NOTE]
```

> ^Skip this network peering step if you installed AKS into a subnet on the Azure Managed Lustre virtual network.

Reviewer 2: ```suggestion
> Skip this network peering step if you installed AKS into a subnet on the Azure Managed Lustre virtual network.
```

> The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the Properties blade and finding the Infrastructure resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_&lt;aks-rg-name&rt;\_&lt;aks-cluster-name&rt;\_&lt;region&rt;**.

Reviewer 2: ```suggestion
The AKS virtual network is created in a separate resource group from the AKS cluster's resource group. You can find the name of this resource group by going to your AKS cluster in the Azure Portal choosing the **Properties** blade and finding the **Infrastructure** resource group.  This resource group contains the virtual network that needs to be paired with the Azure Managed Lustre virtual network. It matches the pattern **MC\_&lt;aks-rg-name&rt;\_&lt;aks-cluster-name&rt;\_&lt;region&rt;**.
```

Author: incorporated

Consult [Virtual Network Peering](/azure/virtual-network/virtual-network-peering-overview) to peer the AKS virtual network with your Azure Manages Lustre virtual network.

> ^[TIP]

Reviewer 2: ```suggestion
> [!TIP]
```

> ^Due to the naming of the MC_ resource groups and virtual networks names of networks can be similar or the same across multiple AKS deployments. When setting up peering pay close attention that you are choosing the AKS networks that you intend to choose.

Reviewer 2: ```suggestion
> Due to the naming of the MC_ resource groups and virtual networks, names of networks can be similar or the same across multiple AKS deployments. When setting up peering pay close attention that you are choosing the AKS networks that you intend to choose.
```

## Connect to the AKS cluster

Connect to the Azure Kubernetes Service cluster by doing these steps:

> 1. Open a terminal session on with access to the Azure CLI tools and login to your Azure Account.

Reviewer 2: ```suggestion
1. Open a terminal session with access to the Azure CLI tools, and login to your Azure account.
```

   ```azurecli
   az login
   ```

1. Sign in to [the Azure portal](https://portal.azure.com).

> 1. Find your AKS cluster and load the **Overview** blade, select the **Connect** button then copy the command for **Download cluster credentials**.

Reviewer 2: ```suggestion
1. Find your AKS cluster. Select the **Overview** blade, then select the **Connect** button and copy the command for **Download cluster credentials**.
```
